src/transformers/models/beit/image_processing_beit.py

```diff
@@ -0,0 +1,519 @@
+
+import numpy as np
+
+from transformers.utils import is_torch_available, is_torch_tensor, is_vision_available
+from transformers.utils.generic import TensorType
+
+from ...image_processing_utils import BaseImageProcessor, BatchFeature, get_size_dict
+from ...image_transforms import center_crop, normalize, rescale, resize, to_channel_dimension_format
+from ...image_utils import (
+    IMAGENET_STANDARD_MEAN,
+    IMAGENET_STANDARD_STD,
+    ChannelDimension,
+    ImageInput,
+    PILImageResampling,
+    is_batched,
+    to_numpy_array,
+    valid_images,
+)
+from ...utils import logging
+
+
+if is_vision_available():
+    import PIL
+
+if is_torch_available():
+    import torch
+
+
+logger = logging.get_logger(__name__)
+
+
+class BeitImageProcessor(BaseImageProcessor):
+    r"""
+    Constructs a DeiT image processor.
+
+    Args:
+        do_resize (`bool`, *optional*, defaults to `True`):
+            Set the class default for the `do_resize` parameter. Controls whether to resize the image's (height, width)
+            dimensions to the specified `size`.
+        size (`Dict[str, int]` *optional*, defaults to `{"height": 256, "width": 256}`):
+            Set the class default for the `size` parameter. Size of the output image after resizing.
+        resample (`PILImageResampling`, *optional*, defaults to `PILImageResampling.BICUBIC`):
+            Set the class default for `resample`. Defines the resampling filter to use if resizing the image.
+        do_center_crop (`bool`, *optional*, defaults to `True`):
+            Set the class default for `do_center_crop`. Controls whether to center crop the image. If the input size is
+            smaller than `crop_size` along any edge, the image is padded with 0's and then center cropped.
+        crop_size (`Dict[str, int]`, *optional*, defaults to `{"height": 224, "width": 224}`):
+            Set the class default for the `crop_size` parameter. Desired output size when applying center-cropping.
+            Only has an effect if `do_center_crop` is set to `True`.
+        do_rescale (`bool`, *optional*, defaults to `True`):
+            Set the class default for the `do_rescale` parameter. Controls whether to rescale the image by the
+            specified scale `rescale_factor`.
+        rescale_factor (`int` or `float`, *optional*, defaults to `1/255`):
+            Set the class default for `rescale_factor`. Defines the scale factor to use if rescaling the image.
+        do_normalize (`bool`, *optional*, defaults to `True`):
+            Set the class default for `do_normalize`. Controls whether to normalize the image.
+        image_mean (`float` or `List[float]`, *optional*, defaults to `IMAGENET_STANDARD_MEAN`):
+            Set the class default for `image_mean`. This is a float or list of floats of length of the number of
+            channels for
+        image_std (`float` or `List[float]`, *optional*, defaults to `IMAGENET_STANDARD_STD`):
+            Image standard deviation.
```

Comment:
For consistency, we could also use Set the class default for `image_std` etc here.